fl4health/parameter_exchange/layer_exchanger.py

```diff
@@ -92,88 +91,76 @@ def pull_parameters(self, parameters: NDArrays, model: nn.Module, config: Option
         model.load_state_dict(current_state, strict=True)
 
 
-class NormDriftParameterExchanger(ParameterExchangerWithPacking[List[str]]):
+class DynamicLayerExchanger(PartialParameterExchanger[List[str]]):
     def __init__(
         self,
+        layer_selection_function: Callable[..., Tuple[NDArrays, List[str]]],
         norm_threshold: float,
         exchange_percentage: float = 0.1,
         normalize: bool = True,
         filter_by_percentage: bool = True,
+        select_drift_more: bool = True,
     ) -> None:
         """
         This exchanger selects those parameters that at the end of each training round, drift away (in l2 norm)
         from their initial values (at the beginning of the same round) by more than self.threshold.
         """
+        self.set_layer_selection_function(layer_selection_function)
         self.parameter_packer = ParameterPackerWithLayerNames()
         assert 0 < exchange_percentage <= 1
         assert 0 < norm_threshold
-        self.threshold = norm_threshold
+        self.norm_threshold = norm_threshold
         self.exchange_percentage = exchange_percentage
         self.set_normalization_mode(normalize)
         self.set_filter_mode(filter_by_percentage)
+        self.set_select_drift_more(select_drift_more)
+
+    def set_layer_selection_function(
+        self, layer_selection_function: Callable[..., Tuple[NDArrays, List[str]]]
+    ) -> None:
+        self.layer_selection_function = layer_selection_function
 
     def set_filter_mode(self, filter_by_percentage: bool) -> None:
         self.filter_by_percentage = filter_by_percentage
 
     def set_normalization_mode(self, normalize: bool) -> None:
         self.normalize = normalize
 
+    def set_select_drift_more(self, select_drift_more: bool) -> None:
+        self.select_drift_more = select_drift_more
+
     def _calculate_drift_norm(self, t1: torch.Tensor, t2: torch.Tensor) -> float:
```

Comment:
Is this method needed anymore? I don't see it being used anywhere

Reply:
I also see there is a functional counterpart `parameter_selection_criteria.py`

Reply:
Yes currently these setters are not used. David also made a similar comment, see:

https://github.com/VectorInstitute/FL4Health/pull/103#discussion_r1476695267

Reply:
Just an update: I've removed all the setters for both `SparseCooParameterExchanger` and `DynamicLayerExchanger`